### PR TITLE
Removing ingress for role-service

### PR DESCRIPTION
**Pull request by xuluna**
# Description
This PR removes the role-service ingress, adds role-service handlers to the proxy-server, updates karavictl to connect to the proxy-server instead of the role-service, and adds telemetry updates.

Proxy-Server changes:

New service handler at internal/proxy/service_handler.go to handle these paths for karavictl:

POST /proxy/roles/ (create a role, role info is in request body)
PATCH /proxy/roles/ (update a role, role info is in request body)
GET /proxy/roles/ (list or get a role, role name is in query parameter)
DELETE /proxy/roles/ (delete a role, role info is in request body)
The handlers for each path will parse the request and make the appropriate grpc request to the role service.

Role-service changes:

Adds grpc server interceptors which will automatically create tracing spans for incoming requests.
Adds telemetry middleware at internal/role-service/middleware/role_telemetry.go to log the time taken for the request, log request info, and set the request info in the span

Most if not all code is similar to storage/tenant-service changes.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/725|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Powerflex E2E Test, and manual test (Helm and RPM)

### CODE REVIEW

File: cmd/karavictl/cmd/api/api_test.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func TestAPI(t *testing.T) {
 		values := url.Values{
 			"key": []string{"value"},
 		}
-		err = insecureClient.Delete(context.Background(), "/delete", nil, values, &resp)
+		err = insecureClient.Delete(context.Background(), "/delete", nil, values, nil, &resp)
```
Comment (alikdell) [inline cmd/karavictl/cmd/api/api_test.go:113]: isn't url should be /proxy/role ?

**Review by alikdell — DISMISSED** [review]

File: internal/role-service/middleware/role_telemetry.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+// List wraps List
+func (t *TelemetryMW) List(ctx context.Context, req *pb.RoleListRequest) (*pb.RoleListResponse, error) {
+	now := time.Now()
+	defer t.timeSince(now, "List")
+
+	span := trace.SpanFromContext(ctx)
+
+	t.log.Info("Getting role")
```
Comment (atye) [inline internal/role-service/middleware/role_telemetry.go:137]: "Listing role"

Comment (xuluna) [reply internal/role-service/middleware/role_telemetry.go]: fixed

File: internal/proxy/role_handler.go [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+		Name: name,
+	})
+	if err != nil {
+		err = fmt.Errorf("getting role %s: %w", name, err)
+		handleJSONErrorResponse(th.log, w, http.StatusInternalServerError, err)
+		return err
+	}
+
+	// return role to client
+	_, err = fmt.Fprint(w, protojson.MarshalOptions{Multiline: true, EmitUnpopulated: true, Indent: ""}.Format(role))
```
Comment (atye) [inline internal/proxy/role_handler.go:217]: You can use the standard json encoding here. This was necessary for getting a tenant to handle empty fields. It shouldn't be necessary for writing the role.

Comment (xuluna) [reply internal/proxy/role_handler.go:217]: I'll change it to `protojson.Format(role)`. storage_handler is using the same MarchalOptions I guess all from the Tenant template.

Comment (xuluna) [reply internal/proxy/role_handler.go]: fixed

**Review by atye — APPROVED** [review]

**Comment by xuluna** [comment]
run e2e test

**Comment by xuluna** [comment]
run e2e test

**Comment by xuluna** [comment]
run e2e test

**Comment by xuluna** [comment]
run e2e test

**Comment by xuluna** [comment]
run e2e test

**Pull request merged**